integration_tests/src/main/python/string_test.py

```diff
@@ -286,6 +286,47 @@ def test_substring():
                 'SUBSTRING(a, 1, NULL)',
                 'SUBSTRING(a, 0, 0)'))
 
+def test_repeat_scalar_and_scalar():
```

Comment:
You should look at the plan produced for this test, but I don't think it is covering what you want.  Spark will do constant folding, which means for most SQL expressions if all of the inputs are literal values, then it will execute it at compile time (on the CPU) and replace it with a literal value that is the result of the computation. If that is the case we can keep the test, but it is not really doing much and we should at least document it, but I personally would remove it.

Reply:
Do you mean such tests are equivalent to something like `assert('abc' == 'abc')`? Is there any way to prevent constant folding? Otherwise, we will not be able to test for those cases.

Reply:
Close to it.  It is not going to be running any of your code.

Reply:
Verify this: None of the code is run with these scalar-scalar tests. So I'm going to remove these tests.